enumerables.rb

```diff
@@ -0,0 +1,147 @@
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  def my_each(&block)
+    return to_enum(:my_each) unless block_given?
+
+    to_a.each(&block)
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    (0...to_a.length).each do |i|
+      yield(to_a[i], i)
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    my_array = []
+    my_each do |item|
+      my_array.push(item) if yield(i)
+    end
+    my_array
+  end
```

Comment:
- [ ] On line 24, you are yielding a variable `i` which you have not defined. Did you mean to yield `item`? Kindly fix this

Reply:
Thank you for your review, I would implement the change and ensure all methods are working @mimipeshy